Pre Prod Logs for Aditya Birla Health Insurance

Formatter opened the pull request.

sourabhsaini11 commented:
@Formatter, kindly get the below issues resolved

common_issues
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format
- for the keys with null values & empty array[] please send a valid value

on_search_1
- Timestamp for /on_search api should be greater than timestamp of /search api
- payments array is missing
- time is missing in items
- descriptor.code should be in a generic enum format(VACCINE_COVER) in items.add_ons
- duplicate provider id: A1 in item[1].add_ons

search_2
- tags are missing in payments
- form.id is missing in xinput

on_select
- Timestamp for /on_search api should be greater than timestamp of /search api
- Quote breakup should only contain the following items: BASE_PRICE, CONVIENCE_FEE, TAX, PROCESSING_FEE 

init
- selected.count must be integer in items
- parent_item_id is missing in items
- virtual_payment_address is missing in payments.params
- BUYER_FINDER_FEES tag group is missing

on_init
- Timestamp for /on_init api should be greater than timestamp of /init api
- Quote breakup should only contain the following items: BASE_PRICE, CONVIENCE_FEE, TAX, PROCESSING_FEE 

confirm
- selected.count must be integer in items
- parent_item_id is missing in items
- virtual_payment_address is missing in payments.params
- tags are missing in payments

on_confirm
- Timestamp for /on_confirm api should be greater than timestamp of /confirm api
- Quote breakup should only contain the following items: BASE_PRICE, CONVIENCE_FEE, TAX, PROCESSING_FEE 

on_status
- Timestamp for /on_status api should be greater than timestamp of /status api
- Quote breakup should only contain the following items: BASE_PRICE, CONVIENCE_FEE, TAX, PROCESSING_FEE 

on_cancel
- Timestamp for /on_cancel api should be greater than timestamp of /cancel api
- Quote breakup should only contain the following items: BASE_PRICE, CONVIENCE_FEE, TAX, PROCESSING_FEE

hiven22157 commented:
@sourabhsaini11  Is it available Discount items in Quote breakup?

Formatter commented:
Hi @sourabhsaini11, thank you for the detailed breakup of issues, please go through the updated logs, I believe we've addressed all of the issues. Please let me know if anything has been missed

sourabhsaini11 commented:
common_issues
- images should be of type array instead of string
- bap_id should be same as the domain part of bap_uri
- instead of qunatity.available, qunatity.selected should be sent post on_search call
- instead of CODI_GENERAL_INFO,tag-group name should be GENERAL_INFO with complete tags

on_search
- payments array is missing

on_search_2
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format
- payments array is missing

select
- form_response should have status & submission_id with valid values
- parent_item_id is missing
- add

on_select
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format
- if add_ons are present, they should be selected as part of select call

init
- virtual_payment_address is missing in params
- status is missing in form_response
- add_ons with id SUPER_CREDIT & PERSONAL_ACCIDENT_COVER are missing in init_2
- init_3 call is missing
- BUYER_FINDER_FEES tag group is missing in payments

on_init
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format
- BUYER_FINDER_FEES tag group is missing in payments
- on_init_3 call is missing


confirm
- add_ons with id SUPER_CREDIT, CLAIM_PROTECT_ & PERSONAL_ACCIDENT_COVER are missing in init_2
- BUYER_FINDER_FEES tag group is missing in payments
- virtual_payment_address is missing in params
- title in quote should be in a generic enum format BASE_PRICE, PROCESSING_FEE, CONVIENCE_FEE, TAX

on_confirm
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format
- id is missing in fulfillments
- created_at, updated_at are missing
- breakup is missing in quote

on_status
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format
- status, created_at & updated_at should have valid values
- id, name & type in fulfillments should have valid values

on_cancel
- Timestamp not in RFC 3339 (YYYY-MM-DDTHH:MN:SS.MSSZ) Format

@Formatter, kindly get the above issues resolved

Formatter commented:
Hi @sourabhsaini11, in the previous logs, a few mandatory fields which didn't have defaults set got skipped as well, so that was a bit of a regression from our end, which has been fixed. 
common issues have been addressed, also please note that init_3 is not applicable in the current flows and hence has been skipped.

anuj-ondc commented:
common issues
  - OFFLINE_CONTRACT tags are missing in SETTLEMENT_TERMS
  - Invalid categories like MULTI_INDIVIDUAL_INSURANCE, FAMILY_FLOTER_INSURANCE
  - Invalid timestamp values, should be greater than the previous call.

  on_search_1
  - Missing GENERAL_INFO tags like COVERAGE_AMOUNT,  ROOM_RENT_CAP etc  & found one extra tag Critical_Illness_Cover 
  - Invalid payments object & missing tags also.

  on_search_2
  - Invalid timestamp
  - Invalid item id, not matching with the one in on_search_1
  - Incorrect tag values for GENERAL_INFO tag group for both items, please check ONDC developer guide page.
  - Incomplete xinput object, missing other form keys, index & headings.
  - Invalid payments object & missing tags also.

  select
  - Empty addons array.
  - Missing form id in xinput.
  - Mising provider id.

  on_select
  - Incorrect fulfillments object.
  - Incorrect GENERAL_INFO tag group values.
  - Missing values for item price.
  - In quote, BASE_PRICE is not mathcing with the item price.
  - Invalid keys like status, created_at, updated_at, id & sourceUrl.

  init_1 
  - Invalid timestamp.
  - Missing form id in xinput.
  - In payments, incorrect values for params amount.

  on_init_1
  - Missing id in fulfillments.
  - Invalid fulfillments state.
  - In payment tags, duplicate values for BUYER_FINDER_FEES.
  - In xinput found extra form headings, please check ONDC developer guide page.
  - In xinput, missing index.
  - Missing values for item price.
  - Quote values are not matching with the previous call.
  - Invalid keys like status, created_at, updated_at, id & sourceUrl.

  init_2
  - Missing form id in xinput.
  - In payments, incorrect values for params amount.

  on_init_2
  - Missing id in fulfillments.
  - Invalid fulfillments state.
  - In xinput found extra form headings, please check ONDC developer guide page.
  - In xinput, missing index.
  - Missing values for item price.
  - Quote values are not matching with the previous call.
  - Invalid keys like status, created_at, updated_at, id & sourceUrl

  confirm
  - Quote values are not matching with the previous call.
  - Missing form id in xinput.
  - fulfillment id is missing.

  on_confirm
  - In payments, incorrect values for params amount.
  - Item price is not matching.
  - Invalid status value.
  - Missing documents array.

  on_status_KYC_unsolicited
  - xinput should contain form_response object not index, & headings.
  - Invalid keys like status, created_at, updated_at, id & sourceUrl
    
    @Formatter, kindly get the above issues resolved

anuj-ondc commented:
@Formatter, we will setup a call for discussing the errors.

anuj-ondc commented:
on_search_1
- In SETTLEMENT_TERMS incorrect value BAP_OFFLINE_CONTRACT, needs be BPP_OFFLINE_CONTRACT
- In xinput, incorrect value of index max.

on_search_2
- In general_info, CO_PAYMENT, CLAIM_SETTLEMENT_RATIO, PRE_HOSPITALIZATION_COVERAGE_DAYS, POST_HOSPITALIZATION_COVERAGE_DAYS, MATERNITY_COVERAGE, INITIAL_WAITING_PERIOD, CASHLESS_HOSPITALS are not found, please check for all calls.
- value key for ROOM_RENT_CAP is not present.
- No need to send code for extra add_ons like CRITICAL_ILLNESS_COVER, SUPER_CREDIT, CLAIM_PROTECT_ (NON-MEDICAL_EXPENSE_WAIVER), PERSONAL_ACCIDENT_COVER, CHRONIC_CARE, 
  CHRONIC_MANAGEMENT_PROGRAM_(O,  "DURABLE_EQUIPMENT_COVER, Second Medical Opinion for listed Major Illnessr, CANCER_BOOSTER etc.
- In SETTLEMENT_TERMS incorrect value BAP_OFFLINE_CONTRACT, needs be BPP_OFFLINE_CONTRACT.
- In GENERAL_INFO, BASE_PRICE, CONVIENCE_FEE, PROCESSING_FEE & TAX are missing.

on_select
- quote price is incorrect in comparision with breakup elements, should be 13,282.
- orderId should not be present.

on_init_1
- incorrect xinput headings component.
- orderId should not be present.
- In Payments, amount needs to be corrected with updated quote.
- In SETTLEMENT_TERMS incorrect value BAP_OFFLINE_CONTRACT, needs be BPP_OFFLINE_CONTRACT.

on_init_2
- incorrect xinput headings component.
- orderId should not be present.

confirm
- Incorrect value for parent_item_id

on_confirm
- In SETTLEMENT_TERMS  value BPP_OFFLINE_CONTRACT is missing
- Incorrect value for parent_item_id
- In GENERAL_INFO tags, please remove PolicyNumber & ProposalNumber as they will be the part of order id & quote id.
  Also remove ReceiptNo.
- In items, fulfillment_ids is missing.
- In add_ons, the selected count should not be 0, not matching with previous calls. Also remove available count.

@Formatter, please fix above issues.

anuj-ondc commented:
common issues
 In GENERAL_INFO tags, ROOM_CATEGORY is missing.

init_2
- In xinput, formId is not matching with the one in on_init_1

on_onit_2
- Price for add_ons is not matching with the one in quote & items, should be 1228.



FLow_2
search2
- In case of family_insurance where multiple instance of same form required the mimetype needs to be -> text/html-multi from seller side & 
 there will be only one submission id for one form.

init_1
- selected add_ons should be 2, SUPER_CREDIT & CRITICAL_ILLNESS_COVER as returned in the on_select_1 call.

on_init_1
- In items, the selected add_ons not matching with the one in quote object. Found one extra in quote ANNUAL_HEALTH_CHECK_UP.

init_2
- In xinput, formId is not matching with the one in on_init_1

on_init_2
- Quote price is not matching with the breakup prices, check for next calls also.
- Processing fee in on_init_1 is 0 & updated in on_init_2.

@Formatter, please fix above issues.

92shreyansh commented:
common Issue:
  - In payments, incorrect value for BUYER_FINDER_FEES_TYPE should be "percent-annualized".
  - For add_ons like TELE_OPD_CONSULTATION the price should be up to 2 deicmal values, same needs to be for all the price component values.


Flow 2:

  on_confirm:
    - quote ID is not matching with previous calls.



Not a finding but as a Information per se:
  on_init_2:
    - quote price in on_init_1 is 10494 & in on_init_2 it is 10493, also total of breakup is not matching with quote price.
    - payment Amount is not matching with previous calls.

@Formatter, please fix above issues.

Formatter commented:
@92shreyansh 
quote price issue for init_1 and init_2 mismatch will continue, it will usually be under <2rs difference, just internal legacy systems where changes cannot be made for now from ABHI side. All buyer Apps will be provided with relevant documentation and communication regarding the same. 

similar issue with quote id for on_confirm, won't be matching in flow 2 as it gets generated later in the journey, internal Policy Administrative System cannot generate a valid quote ID before data that gets captured in the confirm flow. 

percent annualized and 2 point decimal value consistency has been fixed

sourabhsaini11 commented:
@Formatter please fix the below issues
common_issues
- quote value(10494) should exactly match to summation of breakup(10493.62).

on_search
- tags MANDATORY_ARBITRATION & COURT_JURISDICTION are missing in SETTLEMENT_TERMS tag-group.
- CLAIM_SETTLEMENT_RATIO value should be in ratio format(e.x. 0.96).

select
- selected.count value type should be integer.

on_select
- attribute multiple_sumbissions is missing in xinput

on_init
- cancellation_terms are missing
- attribute multiple_sumbissions is missing in xinput
- tags MANDATORY_ARBITRATION & COURT_JURISDICTION are missing in SETTLEMENT_TERMS tag-group.
- attribute url is missing in payments

init_2
- person is missing in fulfillments

on_init_2
- cancellation_terms are missing
- attribute multiple_sumbissions is missing in xinput
- tags MANDATORY_ARBITRATION & COURT_JURISDICTION are missing in SETTLEMENT_TERMS tag-group.

on_confirm
- cancellation_terms are missing

on_init_2
- mime_type must be application/html if submission_id will be sent through /on_status_payment_unsolicited call

sourabhsaini11 commented:
@Formatter , it's a gentle reminder for fixing the above mentioned issues.We will be marking this thread as CLOSED if no such activity is recorded for the next 2 upcoming weeks

JyotiSharma0711 commented:
Please proceed with Pramaan testing at https://pramaan.ondc.org/ for the pre-production environment. Once you obtain a clean report from Pramaan, simply upload it—there is no need for a GitHub PR log review for the Personal Loan use case.

Kindly note that this PR is scheduled to be closed within the next 1 to 2 weeks. Please ensure that any required changes from your side are addressed within this timeframe.

JyotiSharma0711 commented:
Reminder:

Kindly note that this PR is scheduled to be closed within the next 1 to 2 weeks. Please ensure that any required changes from your side are addressed within this timeframe.
